app/views/admin/products/_form.html.erb

```diff
@@ -1,4 +1,4 @@
-<%= form_for(product) do |f| %>
+<%= form_for product, url: url,  method: method do |f| %>
```

Comment:
Interesting solution and I guess this works, allowing the partial to path in the locals for the form.   This can be achieved by passing `[:admin, product]` to form_for and it will handle the correct path helpers.